test/inductor/test_fused_attention.py

```diff
@@ -232,10 +232,14 @@ def sfdp_pattern_5_v2(query, key, value):
             )
             return attn_weight @ value
 
-        self._check_common(sfdp_pattern_5_v1, contains=False)
-        self._check_common(checkpoint_wrapper(sfdp_pattern_5_v1), contains=False)
-        self._check_common(sfdp_pattern_5_v2, contains=False)
-        self._check_common(checkpoint_wrapper(sfdp_pattern_5_v2), contains=False)
+        self._check_common(sfdp_pattern_5_v1, contains=False, check_train=False)
+        self._check_common(
+            checkpoint_wrapper(sfdp_pattern_5_v1), contains=False, check_train=False
+        )
+        self._check_common(sfdp_pattern_5_v2, contains=False, check_train=False)
+        self._check_common(
+            checkpoint_wrapper(sfdp_pattern_5_v2), contains=False, check_train=False
+        )
 
```

Comment:
Hi @eellison @jansel,

I'm trying to replace a pattern with a custom fusions, but only for inference on machines with some specific ISAs.
I had to disable `check_train` for this pattern's UT because [if I try to add a new inference-only pattern corresponding to an existing pattern](https://github.com/sanchitintel/pytorch/compare/master...inference_only_patterns), they don't work because the Inductor [pattern-matcher doesn't allow registration of duplicate patterns](https://github.com/pytorch/pytorch/blob/main/torch/_inductor/pattern_matcher.py#L1095).

So, an alternative I'm thinking of using a separate Inductor pass similar to [`_sfdp_init`](https://github.com/pytorch/pytorch/blob/main/torch/_inductor/fx_passes/joint_graph.py#L34) in `torch/_inductor/fx_passes/joint_graph.py`, that would only be run in case of inference (I can't simply add an argument in `_sfdp_init()` that corresponds to inference-mode because in that case, it'd not generate the serialized patterns that already exist in `torch/_inductor/fx_passes/serialized_patterns`).

Please advise if such an alternative is reasonable. One corner-case is that since [`lazy_init` in `joint_graph.py`](https://github.com/pytorch/pytorch/blob/main/torch/_inductor/fx_passes/joint_graph.py#L28)  only seems to be run once over the lifetime of a PyTorch process, so while it wouldn't cause an issue in deployment (this feature would not be enabled by default, but users would have to specify a custom torch.compile config), it can be problematic to users if they'd try using this feature for inference, and also try using the default Inductor passes in the same process (which doesn't seem like a typical use-case, but this feature would then require a separate UT file as well).

Thanks!

Reply:
I've added the aforementioned changes. Thanks!
The only downside is having a few duplicate patterns in `torch/_inductor/fx_passes/serialized_patterns`, but those files are not duplicate, i.e. each float inference pattern is present in two files, one of which is specific to the custom pass I'm adding.